scripts/Mob.gd

```diff
@@ -1,38 +1,28 @@
-extends KinematicBody2D
+extends Entity
 
 
-const ACCELERATION = 256
-const MAX_SPEED = 32
-const GRAVITY = 300
-
-var direction = 1
-var motion = Vector2.ZERO
-
 func _ready():
+	set_physics_process(true)
 	$AnimatedSprite.play("walk")
 
-func process_horizontal_movement(delta):
-	motion.x += direction * ACCELERATION * delta
-	motion.x = clamp(motion.x, -MAX_SPEED, MAX_SPEED)
-	
-	if $Ledge.is_colliding() == false:
-		direction = direction * -1
-		$Ledge.position.x *= -1
-	
-	if is_on_wall() == true:
-		direction = direction * -1
-		$Ledge.position.x *= -1
-	
-func process_vertical_movement(delta):
-	motion.y += GRAVITY * delta
-	
+func change_direction():
+	direction = direction * -1
+	$Ledge.position.x *= -1
+	$WallCollision.position.x *= -1
+	$WallCollision2.position.x *= -1
```

Comment:
`WallCollision2` is not a good name. It should be more self-explanatory.

Reply:
Will be done in the next merge